web/client/hooks/usePluginItems.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useRef, useMemo } from 'react';
+import join from 'lodash/join';
+import { getConfiguredPlugin } from '../utils/PluginsUtils';
+
+/**
+ * hook to load and configure items in plugins container
```

Comment:
can you explain better what this does. 
It looks to apply to item `Component` some attributes spreading config, adding pluginCfg as object, the usage is not clear.